transports/transports_test.go

```diff
@@ -0,0 +1,21 @@
+package transports
+
+import (
+	"testing"
+
+	"github.com/containers/image/directory"
+	"github.com/containers/image/storage"
+	"github.com/containers/image/transports"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTransportFromImageName(t *testing.T) {
```

Comment:
Having this test is great, but
```
import cycle not allowed in test
package github.com/containers/image/transports (test)
	imports github.com/containers/image/directory
	imports github.com/containers/image/transports
```
Maybe move the test to the c/image/transports/alltransports subpackage.  (Moving the implementation in there as well would also be acceptable; this function does not logically require the alltransports subpackage to be useful, OTOH skopeo imports it anyway, so it makes no difference for it.)

Or, I guess, maybe just drop the test altogether.

Reply:
I moved the code to alltransports.